Source/DataModels/Components/UI/ComponentVideo.cpp

```diff
@@ -0,0 +1,334 @@
+
+	videoStreamIndex = av_find_best_stream(formatCtx, AVMEDIA_TYPE_VIDEO, -1, -1, nullptr, 0);
+	if (videoStreamIndex < 0)
+	{
+		LOG_ERROR("Couldn't find valid video stream inside file.");
+		return;
+	}
+
+	// Find an appropiate video decoder
+	videoCodecParams = formatCtx->streams[videoStreamIndex]->codecpar;
+	videoDecoder = avcodec_find_decoder(videoCodecParams->codec_id);
+	if (!videoDecoder)
+	{
+		LOG_ERROR("Couldn't find valid video decoder.");
+		return;
+	}
+
+	// Set up a video codec context for the decoder
+	videoCodecCtx = avcodec_alloc_context3(videoDecoder);
+	if (!videoCodecCtx)
+	{
+		LOG_ERROR("Couldn't allocate AVCodecContext.");
+		return;
+	}
+	if (avcodec_parameters_to_context(videoCodecCtx, videoCodecParams) < 0)
+	{
+		LOG_ERROR("Couldn't initialise AVCodecContext.");
+		return;
+	}
+	if (avcodec_open2(videoCodecCtx, videoDecoder, nullptr) < 0)
+	{
+		LOG_ERROR("Couldn't open video codec.");
+		return;
+	}
+
+	// Set video parameters and Allocate frame buffer
+	frameWidth = videoCodecParams->width;
+	frameHeight = videoCodecParams->height;
+	frameData = new uint8_t[frameWidth * frameHeight * 4];
+	SetVideoFrameSize(frameWidth, frameHeight);
+	memset(frameData, 0, frameWidth * frameHeight * 4);
+
+	// Allocate memory for packets and frames
+	avPacket = av_packet_alloc();
+	if (!avPacket)
+	{
+		LOG_ERROR("Couldn't allocate AVPacket.");
+		return;
+	}
+	avFrame = av_frame_alloc();
+	if (!avFrame)
+	{
+		LOG_ERROR("Couldn't allocate AVFrame.");
+		return;
+	}
+	initialized = true;
+}
+
+void ComponentVideo::ReadVideoFrame()
+{
+	if (initialized)
+	{
+		int response = -1;
+		int error = 0;
+		while (error >= 0)
+		{
+			error = av_read_frame(formatCtx, avPacket);
+
+			if (avPacket->stream_index != videoStreamIndex)
+			{
+				av_packet_unref(avPacket);
+				continue;
+			}
+
+			// SEEK to frame 0 -> Restart the video timestamp
+			if (error == AVERROR_EOF)
+			{
+				finished = true;
+				if (loop)
+				{
+					RestartVideo();
+					av_packet_unref(avPacket);
+				}
+				if(rotateVertical)
+				{
+					avFrame->data[0] += avFrame->linesize[0] * (videoCodecCtx->height - 1);
+					avFrame->linesize[0] *= -1;
+					avFrame->data[1] += avFrame->linesize[1] * (videoCodecCtx->height / 2 - 1);
+					avFrame->linesize[1] *= -1;
+					avFrame->data[2] += avFrame->linesize[2] * (videoCodecCtx->height / 2 - 1);
+					avFrame->linesize[2] *= -1;
+				}
+
+				break;
+			}
+
+			response = avcodec_send_packet(videoCodecCtx, avPacket);
+			if (response < 0)
+			{
+				LOG_ERROR("Failed to decode packet");
+				// LOG_ERROR("Failed to decode packet: %s.", libav_err2str(response));
+				return;
+			}
+
+			response = avcodec_receive_frame(videoCodecCtx, avFrame);
+			if (response == AVERROR(EAGAIN) || response == AVERROR_EOF)
+			{
+				av_packet_unref(avPacket);
+				continue;
+			}
+			if (response < 0)
+			{
+				LOG_ERROR("Failed to decode frame");
+				// LOG_ERROR("Failed to decode frame: %s.", libav_err2str(response));
```

Comment:
If you want to, you can still add this information in the log.
```suggestion
				LOG_ERROR("Failed to decode frame: {}", libav_err2str(response));
```

Reply:
It is a deprecated function libav_err2str.